harvester/harvester
# Extend upgrade image preloading job deadline from 15 minutes to 1 hour for slower hardware

**Problem:**

The image preloading phase (almost always) expires when the node has a slow disk. The default 15 minutes is just too short.

**Solution:**


A new version of the system-upgrade-controller that is included in the embedded Rancher in the 1.2, 1.3, and development branches of Harvester has an addition to its Plan spec that allows configuring the job deadline. Use this to extend it to an hour.

This PR also upgrades the github.com/rancher/system-upgrade-controller/pkg/apis module to the version that the embedded Rancher in Harvester uses so that we can configure the job deadline. 

**Related Issue:** https://github.com/harvester/harvester/issues/2894

**Test plan:**


1. Deploy Harvester 1.2-head
2. Deploy a build of Harvester with this patch (you may need to update the harvester deployment to fake a version):

```diff
      containers:
      - env:
+       - name: HARVESTER_SERVER_VERSION
+         value: 1.2.2
```

3. Start an upgrade to 1.3.0
4. When your node enters the images preloading phase, quickly find the job:

```console
harvester-qm4x8:~ # kubectl get jobs -n cattle-system
NAME                                                              COMPLETIONS   DURATION   AGE
apply-hvst-upgrade-7v8c6-prepare-on-harvester-qm4x8-with-cfab30   0/1           6m54s      6m54s
```

5. Confirm that its deadline is now set to 1 hour:

```console
harvester-qm4x8:~ # kubectl get job apply-hvst-upgrade-7v8c6-prepare-on-harvester-qm4x8-with-cfab30 -n cattle-system -o yaml | yq .spec.activeDeadlineSeconds
3600
```
